src/Tribe/Attendee_Repository.php

```diff
@@ -1307,4 +1307,223 @@ protected function maybe_send_attendee_email( $attendee_id, $attendee_data ) {
+			$override = $loop === 0;
+
+			switch ( $order_by ) {
+				case 'security_code':
+					$this->order_by_security_code( $order, $after, $override );
+					break;
+				case 'check_in':
+					$this->order_by_check_in( $order, $after, $override );
+					break;
+				case 'rsvp_status':
+					$this->order_by_rsvp_status( $order, $after, $override );
+					break;
+				case '__none':
+					unset( $this->query_args['orderby'] );
+					unset( $this->query_args['order'] );
+					break;
+				default:
+					$after = $after || 1 === $loop;
+					if ( empty( $this->query_args['orderby'] ) ) {
+						// In some versions of WP, [ $order_by, $order ] doesn't work as expected. Using explict value setting instead.
+						$this->query_args['orderby'] = $order_by;
+						$this->query_args['order']   = $order;
+					} else {
+						$add = [ $order_by => $order ];
+						// Make sure all `orderby` clauses have the shape `<orderby> => <order>`.
+						$normalized = [];
+
+						if ( ! is_array( $this->query_args['orderby'] ) ) {
+							$this->query_args['orderby'] = [
+								$this->query_args['orderby'] => $this->query_args['order']
+							];
+						}
+
+						foreach ( $this->query_args['orderby'] as $k => $v ) {
+							$the_order_by                = is_numeric( $k ) ? $v : $k;
+							$the_order                   = is_numeric( $k ) ? $default_order : $v;
+							$normalized[ $the_order_by ] = $the_order;
+						}
+						$this->query_args['orderby'] = $normalized;
+						$this->query_args['orderby'] = array_merge( $this->query_args['orderby'], $add );
+					}
+					break;
+			}
+		}
+	}
+
+	/**
+	 * Sets up the query filters to order attendees by the security code meta.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $order      The order direction, either `ASC` or `DESC`; defaults to `null` to use the order
+	 *                           specified in the current query or default arguments.
+	 * @param bool   $after      Whether to append the duration ORDER BY clause to the existing clauses or not;
+	 *                           defaults to `false` to prepend the duration clause to the existing ORDER BY
+	 *                           clauses.
+	 * @param bool   $override   Whether to override existing ORDER BY clauses with this one or not; default to
+	 *                           `true` to override existing ORDER BY clauses.
+	 */
+	protected function order_by_security_code( $order = null, $after = false, $override = true ) {
+		global $wpdb;
+
+		$meta_alias     = 'security_code';
+		$meta_keys_in   = $this->prepare_interval( $this->security_code_keys() );
+		$postmeta_table = "orderby_{$meta_alias}_meta";
+		$filter_id      = 'order_by_security_code';
+
+		$this->filter_query->join(
+			"
+			LEFT JOIN {$wpdb->postmeta} AS {$postmeta_table}
+				ON (
+					{$postmeta_table}.post_id = {$wpdb->posts}.ID
+					AND {$postmeta_table}.meta_key IN {$meta_keys_in}
+				)
+			"
+			,
+			$filter_id,
+			true
+		);
+
+		$order = $order === null
```

Comment:
Maybe document a bit this ternary so that it can be more clear what's doing?

Reply:
I see that we're using this one all the time, maybe we can have a small method with parameters. If we choose a good name for the method it'd be clear what's doing out of the box.